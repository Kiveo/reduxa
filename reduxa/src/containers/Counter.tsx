import React from 'react';

const buttonStyle = {
  padding: '1rem 2rem',
  margin: '0.25rem',
}

const Counter = (props: any) => {
  return (
    <div id="counter">
      <p>Global Count is: <span id="globalCount">{props.count || 0}</span></p>
      <p>Global State Controls</p>
      <button type="button" id="addButton" style={buttonStyle} onClick={() => props.dispatch(props.increment())}>ADD</button>
      {/* // ! minimizing scope for testing redux and redux coverage report */}
      {/* <button type="button" style={buttonStyle} onClick={() => props.dispatch(props.incrementByAmount(2))}>ADD 2</button>
      <button type="button" style={buttonStyle} onClick={() => props.dispatch(props.incrementAsync(10))}>Delayed ADD 10</button>
      <button type="button" style={buttonStyle} onClick={() => props.dispatch(props.decrement())}>SUBTRACT</button>
      <button type="button" style={buttonStyle} onClick={() => props.dispatch(props.decrementAsync(10))}>Delayed Reduce 10</button> */}

      {/* <hr style={{ border: '1px solid black' }} />
      <p>Local Value is: <span id="localCount" data-test="localCount">{props.counter || 0}</span></p>
      <p>Local State Controls</p>
      <button type="button" style={buttonStyle} onClick={() => props.localIncrement()}>Increase Local Value</button>
      <button type="button" style={buttonStyle} onClick={() => props.localDecrement()}>Reduce Local Value</button> */}
    </div>
  )
}

export default Counter;