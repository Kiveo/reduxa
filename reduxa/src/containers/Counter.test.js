import React from 'react';
import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import Counter from './Counter';
import configureStore from 'redux-mock-store';

const middlewares = []
const mockStore = configureStore(middlewares)

describe('Counter Component Tests', () => {
  it('renders', function () {
    const wrapper = shallow(<Counter />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should update local Count onClick of ADD', async () => {
    // -- arrange --
    // const dummyStore = mockStore({ count: 2 })
    let sampleCount = 1986;
    const mockAdd = jest.fn()
    const mockDispatch = jest.fn()
    const wrapper = shallow(<Counter
      count={sampleCount}
      increment={mockAdd}
      dispatch={mockDispatch}
    />);
    expect(wrapper.find('#globalCount').text()).toBe('1986'); // default value
    expect(mockAdd).not.toHaveBeenCalled();
    expect(mockDispatch).not.toHaveBeenCalled();

    // -- act --
    const addButton = wrapper.find('#addButton');
    expect(addButton).toHaveLength(1);
    addButton.props().onClick();
    wrapper.update();

    // -- assert --
    expect(mockAdd).toHaveBeenCalled();
    expect(mockDispatch).toHaveBeenCalled();
    // expect(wrapper.find('#globalCount').text()).toBe('1');
  });



});
