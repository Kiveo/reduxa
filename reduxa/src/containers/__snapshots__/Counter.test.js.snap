// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Tests renders 1`] = `ShallowWrapper {}`;
